server/src/main/java/io/deephaven/server/console/completer/PythonAutoCompleteObserver.java

```diff
@@ -33,12 +49,29 @@ public class PythonAutoCompleteObserver extends SessionCloseableObserver<AutoCom
      * We only log timing for completions that take longer than, currently, 100ms
      */
     private static final long HUNDRED_MS_IN_NS = 100_000_000;
-    private final Provider<ScriptSession> scriptSession;
 
-    public PythonAutoCompleteObserver(StreamObserver<AutoCompleteResponse> responseObserver,
-            Provider<ScriptSession> scriptSession, final SessionState session) {
+    /** Track parsers by their session state, to ensure each session has its own, singleton, parser */
+    private static final Map<SessionState, JediCompleter> parsers = Collections.synchronizedMap(new WeakHashMap<>());
```

Comment:
extremely unsafe to do this.

jedi is NOT threadsafe and should NOT be accessed by more than one caller at a time.
I plan to eventually put in some kind of threading and locking that ensures we don't accidentally use it in a parallelized manner, and having only a single completer would help with that...